src/tracer_options.cpp

```diff
@@ -3,25 +3,108 @@
 #include <iterator>
 #include <sstream>
 
+#include <datadog/tags.h>
+#include <opentracing/ext/tags.h>
+
 namespace ot = opentracing;
 
 namespace datadog {
 namespace opentracing {
 
+// Extracts key-value pairs from a string.
+// Duplicates are overwritten.
+// Intended use is for settings tags from DD_TAGS options.
+std::map<std::string, std::string> keyvalues(std::string text, char itemsep, char tokensep,
+                                             char escape) {
+  // early-return if empty
+  if (text.empty()) {
+    return {};
+  }
+
+  bool esc = false;
+  std::map<std::string, std::string> kvp;
+  std::string key;
+  std::string val;
+  bool keyfound = false;
+  auto assignchar = [&](char c) {
+    if (keyfound) {
+      val += c;
+    } else {
+      key += c;
+    }
+    esc = false;
+  };
+  auto addkv = [&](std::string key, std::string val) { kvp[key] = val; };
+  for (auto ch : text) {
+    if (esc) {
+      assignchar(ch);
+      continue;
+    }
+    if (ch == escape) {
+      esc = true;
+      continue;
+    }
+    if (ch == tokensep) {
+      addkv(key, val);
```

Comment:
we should probably do minimal validation that key and value are not empty?

I _know_ not having an empty key should definitely be here... but... empty value?

```
DD_TAGS=one:,two:,three:
```

but also

```
DD_TAGS=:,:,:,:
```